src/core/buffers/representation/append_data.ts

```diff
@@ -49,8 +49,12 @@ export default function appendDataToSourceBufferWithRetries<T>(
 
   return append$.pipe(
     catchError((appendError : Error) => {
-      if (!appendError || appendError.name !== "QuotaExceededError") {
-        throw new MediaError("BUFFER_APPEND_ERROR", appendError.toString(), true);
+      if (appendError && appendError.name !== "QuotaExceededError") {
+        const reason = appendError instanceof Error ?
+          appendError.toString() :
+          "Couldn't append buffer";
```

Comment:
Why do we do not throw an error from the QueuedSourceBuffer in the first place?

It would be preferable to have a more precize error closer to where it happened.